chessence-frontend/src/api/socket.js

```diff
@@ -0,0 +1,8 @@
+import { io } from "socket.io-client";
+
+// setting URL to "undefined" means the URL
+// will be computed from the `window.location` object
+const URL = "http://localhost:4000";
```

Comment:
LGTM! Is this just to test it in a separate window, or is it to separate the host on localhost:3000 and the person joining the room on localhost:4000?

Reply:
the frontend is on :3000; the backend has two ports, :8000 for the REST API, and :4000 for the web socket.